### PR TITLE
Faster serialization of Avro unions

### PR DESCRIPTION
**Pull request by piotrp**
## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [x] Code is cleaned from temporary changes and commented out lines
- [x] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [x] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

File: utils/schemed-kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/schemedkafka/encode/ToAvroSchemaBasedEncoder.scala [hunk 1]
```diff
@@ -150,10 +150,13 @@ class ToAvroSchemaBasedEncoder(avroSchemaEvolution: AvroSchemaEvolution, validat
...
-    if (!schema.hasEnumSymbol(symbol))
-      error(s"Not expected symbol: $symbol for field: $fieldName with schema: $schema")
-    else
+  private def encodeEnum(symbol: String, schema: Schema, fieldName: Option[String]): WithError[EnumSymbol] =
+    if (!schema.hasEnumSymbol(symbol)) {
+      val allowedEnumValues = schema.getEnumSymbols.asScala.mkString(", ")
+      error(
+        s"Not expected symbol: $symbol for field: $fieldName with schema: ${schema.getFullName}, allowed values: $allowedEnumValues"
+      )
+    } else
```
Comment (arkadius) [inline utils/schemed-kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/schemedkafka/encode/ToAvroSchemaBasedEncoder.scala:159]: Please add curly braces after `else`

File: docs/Changelog.md [hunk 2]
```diff
@@ -45,6 +45,7 @@
 * [#6826](https://github.com/TouK/nussknacker/pull/6826) Security fix: added validation of expression used inside 
   indexer for Maps and Lists (for example `{1,2,3}[#otherList.remove(1) == null ? 0 : 0]`). This allowed executing 
   some types of unallowed expressions.
+* [#6880](https://github.com/TouK/nussknacker/pull/6880) Performance optimization of generating Avro messages with unions
```
Comment (arkadius) [inline docs/Changelog.md:48]: Maybe let's add `- shorter message in logs`? :) It ook me a while to find out where is the optimisation ;)

Comment (piotrp) [reply docs/Changelog.md:48]: Added, along with an explanation in code

File: utils/schemed-kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/schemedkafka/encode/ToAvroSchemaBasedEncoder.scala (hunk 1 above)
Comment (piotrp) [reply utils/schemed-kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/schemedkafka/encode/ToAvroSchemaBasedEncoder.scala]: Added

**Comment by lciolecki** [comment]
What about encodeRecord? There is also $schema in error.

**Comment by piotrp** [comment]
I left full schema in places that are likely to be called directly, not as part of a union. In case of records it may be helpful to have a full schema in error message.

This PR is meant as a quick fix to work around the most important inefficiency: handling schemas with nulls (e.g. `type: ["null", "string"]`). This entire encoder should be rewritten to handle unions efficiently.

**Pull request merged**